src/server/client-data-mapper.js

```diff
@@ -3,12 +3,144 @@ import cloneDeep from '../utils/clone-deep.js';
+
+  return {
+    ...rawTaskParams,
+    updatedOn
+  };
+};
+
+const mapTaskAttempt = rawTaskAttempt => {
+  const startedOn = rawTaskAttempt.startedOn && rawTaskAttempt.startedOn.toISOString();
+  const completedOn = rawTaskAttempt.completedOn && rawTaskAttempt.completedOn.toISOString();
+
+  return {
+    ...rawTaskAttempt,
+    startedOn,
+    completedOn
+  };
+};
+
+const mapTask = rawTask => {
+  const taskParams = rawTask.taskParams && mapTaskParams(rawTask.taskParams);
+  const attempts = rawTask.attempts && rawTask.attempts.map(attempt => mapTaskAttempt(attempt));
+
+  return {
+    ...rawTask,
+    taskParams,
+    attempts
+  };
+};
+
+const mapImportBatch = (rawBatch, rawUser, allowedUserFields) => {
+  const createdOn = rawBatch.createdOn && rawBatch.createdOn.toISOString();
+  const completedOn = rawBatch.completedOn && rawBatch.completedOn.toISOString();
+  const createdBy = mapUser(rawUser, allowedUserFields);
+  const tasks = rawBatch.tasks && rawBatch.tasks.map(task => mapTask(task));
+
+  return {
+    ...rawBatch,
+    createdOn,
+    completedOn,
+    createdBy,
+    tasks
+  };
+};
+
+const mapSection = (section, userMap, allowedUserFields) => {
+  return {
+    ...section,
+    deletedBy: section.deletedBy ? mapUser(userMap.get(section.deletedBy), allowedUserFields) : section.deletedBy
+  };
+};
+
+const mapDocOrRevision = (docOrRevision, userMap, allowedUserFields) => {
+  const result = {};
+
+  for (const [key, value] of Object.entries(docOrRevision)) {
+    switch (key) {
+      case 'createdOn':
+      case 'updatedOn':
+        result[key] = value ? value.toISOString() : value;
+        break;
+      case 'createdBy':
+      case 'updatedBy':
+        result[key] = value ? mapUser(userMap.get(value), allowedUserFields) : value;
+        break;
+      case 'contributors':
+        result[key] = value.map(c => mapUser(userMap.get(c), allowedUserFields));
+        break;
+      case 'sections':
+        result[key] = value.map(s => mapSection(s, userMap, allowedUserFields));
+        break;
+      case 'cdnResources':
+        break;
+      default:
+        result[key] = value;
+        break;
+    }
+  }
+
+  return result;
+};
+
+const mapRoomInvitation = rawInvitation => {
+  const sentOn = rawInvitation.sentOn && rawInvitation.sentOn.toISOString();
+
+  return {
+    ...rawInvitation,
+    sentOn
+  };
+};
+
+const mapLessonSchedule = rawSchedule => {
+  const startsOn = rawSchedule.startsOn && rawSchedule.startsOn.toISOString();
+
+  return {
+    ...rawSchedule,
+    startsOn
+  };
+};
+
+const mapLesson = rawLesson => {
+  const createdOn = rawLesson.createdOn && rawLesson.createdOn.toISOString();
+  const updatedOn = rawLesson.updatedOn && rawLesson.updatedOn.toISOString();
+  const schedule = rawLesson.schedule && mapLessonSchedule(rawLesson.schedule);
+
+  return {
+    ...rawLesson,
+    createdOn,
+    updatedOn,
+    schedule
+  };
+};
```

Comment:
Is there a reason why you implemented all those function outside of the class?

Reply:
Felt like the class was super convoluted 🤔 But I'm not very attached to the idea, I can place them inside the class if you dislike this.

Reply:
Reconsidered, moved everything in the class 👍

Reply:
Ok, I felt the class was a little bit empty before :-)